### ISSUE TITLE
Alignment of 'deleted' lines is misaligned when using tabstop

### ISSUE THREAD

**Issue by andrewferrier** [lewis6991/gitsigns.nvim#595]
### Description

When using gitsigns with `toggle_deleted()` enabled, the indentation of 'deleted' virtual lines is incorrect (too far to the left) when using 'real' tabs rather than spaces (which is necessary for some filetypes like Makefiles)

### Neovim version

NVIM v0.8.0-dev-nightly-2-gea13dce3b

### Operating system and version

Linux lawn 5.18.11-arch1-1

### Expected behavior

The 'deleted counterpart' of a modified line is indented the same amount visually as the modified line itself (assuming the intentation itself is not modified).

### Actual behavior

'Deleted counterpart' lines are shown several columns to the left of where they should be.

Note that setting `vim.opt.number = true` appears to make the problem *worse* (not sure if this helps diagnose the problem).

Same issue occurs with NeoVim 0.7.

### Minimal config

```Lua
vim.o.packpath = '/tmp/nvim/site'

local plugins = {
  gitsigns = 'https://github.com/lewis6991/gitsigns.nvim',
}

for name, url in pairs(plugins) do
  local install_path = '/tmp/nvim/site/pack/test/start/'..name
  if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system { 'git', 'clone', '--depth=1', url, install_path }
  end
end

require('gitsigns').setup{
  debug_mode = true,
}

require('gitsigns').toggle_deleted()

vim.opt.tabstop = 8
vim.opt.number = true
```

### Steps to reproduce

1. `nvim --clean -u minimal.lua`

2. Open any existing files that uses 'real' tabs for indentation (e.g. a GNU Makefile).

3. Modify any existing indented line (but don't touch the indentation), which will trigger a deleted 'shadow'
virtual line to be shown.

4. Notice that the indentation of the shadow line is significantly to the left
of where it should be

### Gitsigns debug messages

```
signs.init: Using vimfn signs
run_job: git --no-pager --literal-pathspecs --version
setup_highlights: Deriving GitSignsAdd from DiffAdd
setup_highlights: Deriving GitSignsChange from DiffChange
setup_highlights: Deriving GitSignsDelete from DiffDelete
setup_highlights: Deriving GitSignsAddNr from GitSignsAdd
setup_highlights: Deriving GitSignsChangeNr from GitSignsChange
setup_highlights: Deriving GitSignsDeleteNr from GitSignsDelete
setup_highlights: Deriving GitSignsAddLn from DiffAdd
setup_highlights: Deriving GitSignsChangeLn from DiffChange
setup_highlights: Deriving GitSignsCurrentLineBlame from NonText
setup_highlights: Deriving GitSignsAddInline from TermCursor
setup_highlights: Deriving GitSignsDeleteInline from TermCursor
setup_highlights: Deriving GitSignsChangeInline from TermCursor
setup_highlights: Deriving GitSignsAddLnInline from GitSignsAddInline
setup_highlights: Deriving GitSignsChangeLnInline from GitSignsChangeInline
setup_highlights: Deriving GitSignsDeleteLnInline from GitSignsDeleteInline
setup_highlights: Deriving GitSignsAddLnVirtLn from GitSignsAddLn
setup_highlights: Deriving GitSignsChangeVirtLn from GitSignsChangeLn
setup_highlights: Deriving GitSignsDeleteVirtLn from DiffDelete
setup_highlights: Deriving GitSignsAddLnVirtLnInLine from GitSignsAddLnInline
setup_highlights: Deriving GitSignsChangeVirtLnInLine from GitSignsChangeLnInline
setup_highlights: Deriving GitSignsDeleteVirtLnInLine from GitSignsDeleteLnInline
fn: Highlight GitSignsAdd is already defined
fn: Highlight GitSignsChange is already defined
fn: Highlight GitSignsDelete is already defined
fn: Highlight GitSignsAddNr is already defined
fn: Highlight GitSignsChangeNr is already defined
fn: Highlight GitSignsDeleteNr is already defined
fn: Highlight GitSignsAddLn is already defined
fn: Highlight GitSignsChangeLn is already defined
fn: Highlight GitSignsCurrentLineBlame is already defined
fn: Highlight GitSignsAddInline is already defined
fn: Highlight GitSignsDeleteInline is already defined
fn: Highlight GitSignsChangeInline is already defined
fn: Highlight GitSignsAddLnInline is already defined
fn: Highlight GitSignsChangeLnInline is already defined
fn: Highlight GitSignsDeleteLnInline is already defined
fn: Highlight GitSignsAddLnVirtLn is already defined
fn: Highlight GitSignsChangeVirtLn is already defined
fn: Highlight GitSignsDeleteVirtLn is already defined
fn: Highlight GitSignsAddLnVirtLnInLine is already defined
fn: Highlight GitSignsChangeVirtLnInLine is already defined
fn: Highlight GitSignsDeleteVirtLnInLine is already defined
run_job: git --no-pager --literal-pathspecs rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
attach(1): Attaching (trigger=BufRead)
run_job: git --no-pager --literal-pathspecs config user.name
run_job: git --no-pager --literal-pathspecs rev-parse --show-toplevel --absolute-git-dir --abbrev-ref HEAD
run_job: git --no-pager --literal-pathspecs --git-dir=/home/ferriera/dotfiles/.git -c core.quotepath=off ls-files --stage --others --exc
lude-standard --eol /home/ferriera/dotfiles/Makefile
watch_gitdir(1): Watching git dir
run_job: git --no-pager --literal-pathspecs --git-dir=/home/ferriera/dotfiles/.git show :0:Makefile
update(1): updates: 1, jobs: 6
update(1): updates: 2, jobs: 6
```

**Comment by lewis6991** [comment]
Upstream issue: https://github.com/neovim/neovim/issues/17148

Duplicate of #445

**Comment by andrewferrier** [comment]
Sorry, didn't see the dup. Thanks for the link.

**Comment by lewis6991** [comment]
We can keep this open for visibility

**Comment by lewis6991** [comment]
Fixed in neovim/neovim#17148

**Comment by andrewferrier** [comment]
Yep, been using that for a while and it's all fixed. Thanks!
